### PR TITLE
Support separated ITenantStore/ITenantProvider

### LINKED ISSUE #94: Support ITenantStore/ITenantProvider split
Corvus.Tenancy now separates read and write.

Consistent with the new property bag way of doing things, modify the tenancy API to update tenants via a PATCH, rather than requiring callers to download the tenant, make changes, then PUT the entire tenant back again.

### PR DESCRIPTION
**Pull request by idg10**
Resolves #94

### CODE REVIEW

File: Solutions/Marain.Tenancy.OpenApi.Service/Marain/Tenancy/OpenApi/TenancyServices.yaml [hunk 1]
```diff
@@ -280,6 +284,41 @@ components:
...
+          description: A JSON-Pointer. Either /name or /properties/propertyName
+        op: 
+          type: string 
+          description: The operation to be performed.
+          # Note: JSON Patch also supports move, copy, and test, but we only support these:
+          enum: [add, replace, remove]
+        value: 
+          description: The value to add or set
+          # We need to set type to none to work around https://github.com/menes-dotnet/Menes/issues/37
+          type: none
```
Comment (mwadams) [inline Solutions/Marain.Tenancy.OpenApi.Service/Marain/Tenancy/OpenApi/TenancyServices.yaml:314]: Could you not just omit `type`?

File: docs/adr/0002-separation-of-read-and-modify.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Separation of read and modify operations
+
+## Status
+
+Proposed.
```
Comment (mwadams) [inline docs/adr/0002-separation-of-read-and-modify.md:5]: Should be "accepted"

File: Solutions/Marain.Tenancy.OpenApi.Service/Marain/Tenancy/OpenApi/TenancyServices.yaml (hunk 1 above)
Comment (idg10) [reply Solutions/Marain.Tenancy.OpenApi.Service/Marain/Tenancy/OpenApi/TenancyServices.yaml:314]: No, that was the first thing I tried. And as it says in that bug, Menes has this code:

```csharp
case null: // Lazy people don't say that they are an object
    return JsonObjectType.Object;
```

This is what kicks in if you don't set a `type`. That's why we need to set it explicitly to `none`.

File: docs/adr/0002-separation-of-read-and-modify.md (hunk 2 above)
Comment (idg10) [reply docs/adr/0002-separation-of-read-and-modify.md]: Done.

**Pull request merged**